File: BE/src/main/java/com/ecommerce/service/HeroBannerService.java
package com.ecommerce.service;

import com.ecommerce.entity.HeroBanner;
import com.ecommerce.repository.HeroBannerRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class HeroBannerService {
    
    @Autowired
    private HeroBannerRepository heroBannerRepository;
    
    /**
     * Get all active hero banners for a specific locale
     * @param locale The locale (e.g., "en", "es")
     * @return List of active hero banners sorted by sortOrder
     */
    public List<HeroBanner> getActiveHeroBanners(String locale) {
        if (locale == null || locale.trim().isEmpty()) {
            locale = "en"; // Default to English
        }
        return heroBannerRepository.findByIsActiveTrueAndLocaleOrderBySortOrder(locale);
    }
    
    /**
     * Get all active hero banners regardless of locale
     * @return List of active hero banners sorted by sortOrder
     */
    public List<HeroBanner> getAllActiveHeroBanners() {
        return heroBannerRepository.findByIsActiveTrueOrderBySortOrder();
    }
    
    /**
     * Get all hero banners for a specific locale (including inactive)
     * @param locale The locale
     * @return List of hero banners sorted by sortOrder
     */
    public List<HeroBanner> getHeroBannersByLocale(String locale) {
        if (locale == null || locale.trim().isEmpty()) {
            locale = "en";
        }
        return heroBannerRepository.findByLocaleOrderBySortOrder(locale);
    }
    
    /**
     * Get a specific hero banner by ID
     * @param id The hero banner ID
     * @return Optional containing the hero banner if found
     */
    public Optional<HeroBanner> getHeroBannerById(String id) {
        return heroBannerRepository.findById(id);
    }
    
    /**
     * Get all hero banners
     * @return List of all hero banners
     */
    public List<HeroBanner> getAllHeroBanners() {
        return heroBannerRepository.findAll();
    }
}

File: FE/packages/web/src/services/heroBannerService.ts
import { API_CONFIG, getCurrentEnvConfig } from '../config/apiConfig';

export interface HeroBanner {
  id: string;
  title: string;
  subtitle: string;
  description: string;
  discount: string;
  price: string;
  buttonText: string;
  image: string;
  bgColor: string;
  isActive: boolean;
  sortOrder: number;
  locale: string;
  createdAt: string;
  updatedAt: string;
}

class HeroBannerService {
  private baseUrl = `${getCurrentEnvConfig().API_BASE_URL}/hero-banners`;

  /**
   * Get all active hero banners for a specific locale
   */
  async getActiveHeroBanners(locale: string = 'en'): Promise<HeroBanner[]> {
    try {
      const response = await fetch(`${this.baseUrl}?locale=${locale}`);
      
      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }
      
      const data = await response.json();
      return data;
    } catch (error) {
      console.error('Error fetching hero banners:', error);
      throw error;
    }
  }

  /**
   * Get all active hero banners regardless of locale
   */
  async getAllActiveHeroBanners(): Promise<HeroBanner[]> {
    try {
      const response = await fetch(`${this.baseUrl}/all`);
      
      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }
      
      const data = await response.json();
      return data;
    } catch (error) {
      console.error('Error fetching all hero banners:', error);
      throw error;
    }
  }

  /**
   * Get a specific hero banner by ID
   */
  async getHeroBannerById(id: string): Promise<HeroBanner> {
    try {
      const response = await fetch(`${this.baseUrl}/${id}`);
      
      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }
      
      const data = await response.json();
      return data;
    } catch (error) {
      console.error(`Error fetching hero banner ${id}:`, error);
      throw error;
    }
  }

  /**
   * Get hero banners by locale (including inactive)
   */
  async getHeroBannersByLocale(locale: string): Promise<HeroBanner[]> {
    try {
      const response = await fetch(`${this.baseUrl}/locale/${locale}`);
      
      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }
      
      const data = await response.json();
      return data;
    } catch (error) {
      console.error(`Error fetching hero banners for locale ${locale}:`, error);
      throw error;
    }
  }
}

const heroBannerService = new HeroBannerService();
export default heroBannerService;
